diff --git a/better_exposed_filters_exposed_form_plugin.inc b/better_exposed_filters_exposed_form_plugin.inc
index 69b38c2..8cefabf 100644
--- a/better_exposed_filters_exposed_form_plugin.inc
+++ b/better_exposed_filters_exposed_form_plugin.inc
@@ -1318,8 +1318,8 @@ Title Desc|Z -> A</pre> Leave the replacement value blank to remove an option al
         $identifier = $form['#info']["filter-$label"]['value'];
         if (!empty($form[$identifier])) {
           // Move exposed operators with exposed filters
-          if (!empty($this->display->display_options['filters'][$identifier]['expose']['use_operator'])) {
-            $op_id = $this->display->display_options['filters'][$identifier]['expose']['operator_id'];
+          if (!empty($filters[$label]->options['expose']['use_operator'])) {
+            $op_id = $filters[$label]->options['expose']['operator_id'];
             $secondary[$op_id] = $form[$op_id];
             unset($form[$op_id]);
           }
